.pre-commit-config.yaml

```diff
@@ -0,0 +1,13 @@
+repos:
```

Comment:
Hi @kellerza 
what is this yaml for?

Reply:
Hi Roman, it is a framework to install git pre-commit hooks (https://pre-commit.com/) that performs a bunch of simple checks before accepting a commit.

Might duplicate some of the other CI checks/tasks you have on github, but codespell really helps for people (myself included) who are too dependent on just typing and Office correcting